Add resource schema-based support for static default values (#674)
hashicorp/terraform-plugin-framework: website/docs/plugin/framework/migrating/attributes-blocks/default-values.mdx

---
page_title: 'Attribute default values: Migrating from SDKv2 to the Framework'
description: >-
  Specify a default when the Terraform configuration does not supply a value for resource attributes.
  Migrate attribute defaults in SDKv2 to AttributePlanModifier in the Framework.
---

# Default Values

> Default values support is only available in the Framework for resources. Default values for data sources and
> providers should be handled within the [data source configure method](/terraform/plugin/framework/data-sources/configure#define-data-source-configure-method)
> or the [provider configure method](/terraform/plugin/framework/providers#configure-method), respectively.

Reviewer: Defaults for data sources should be handled within the `Read` method for now. 😉 (Also minor grammar suggestions for active voice)

```suggestion
Default values support is only available in the Framework for resources. Handle default values for data source attributes within the [data source `Read` method](terraform/plugin/framework/data-sources#read-method) and default values for provider attributes within the [provider `Configure` method](/terraform/plugin/framework/providers#configure-method).
```

# Default Values

Default values set a value for an attribute when the Terraform configuration does not provide one. In SDKv2 and the

> Framework default values are set via the `Default` field on an attribute's schema.

Reviewer: I think it might be good to state upfront here that default value support is only available in the framework for resources and what to do for provider and data source schemas.

The description above needs a tweak as well 👍
